assets/js/lesson_22/to_do_list.js

```diff
@@ -0,0 +1,229 @@
+let ul = document.querySelector('#ul');
+let lastSelected = null;
+
+ul.addEventListener('click', function (e) {
+    if (e.target === this) {
+        return false;
+    }
+
+    if (!e.ctrlKey) {
+        clearSelected(this.children);
+    } else if (e.target.classList.contains('selected')) {
+        e.target.classList.remove('selected');
+        return;
+    }
+
+    if (e.shiftKey) {
+        selectRange(lastSelected, e.target);
+    } else {
+        addSelected(e.target);
+        lastSelected = e.target;
+    }
+})
+
+function clearSelected(elems) {
+    for (let elem of elems) {
+        elem.classList.remove('selected');
+    }
+}
+
+function addSelected(target) {
+    target.classList.add('selected');
+}
+
+function selectRange(start, end) {
+    let startIndex = Array.from(ul.children).indexOf(start);
+    let endIndex = Array.from(ul.children).indexOf(end);
+
+    for (let i = Math.min(startIndex, endIndex); i <= Math.max(startIndex, endIndex); i++) {
+        ul.children[i].classList.add('selected');
+    }
+}
```

Comment:
nice, but Math.min and Math.max need to be wrapped in separate variables

Reply:
Done